This is very much a hack that ought to only mask out a gcc build.

--- a/make/lib/Awt2dLibraries.gmk	Sat Apr 15 10:33:05 2023
+++ b/make/lib/Awt2dLibraries.gmk	Sat Apr 15 09:45:57 2023
@@ -69,7 +69,7 @@
 
 ################################################################################
 
-ifeq ($(call And, $(call isTargetOs, solaris) $(call isTargetCpuArch, sparc)), true)
+ifeq ($(call And, $(call isTargetOs, solaris) $(call isTargetCpuArch, not-really-sparc)), true)
 
   # libmlib_image_v is basically built from mlib_image sources, with some additions
   # and some exclusions.
@@ -128,7 +128,7 @@
     $(TOPDIR)/src/$(MODULE)/$(OPENJDK_TARGET_OS_TYPE)/native/common/awt \
     #
 
-ifeq ($(call And, $(call isTargetOs, solaris) $(call isTargetCpuArch, sparc)), true)
+ifeq ($(call And, $(call isTargetOs, solaris) $(call isTargetCpuArch, not-really-sparc)), true)
   LIBAWT_EXTRA_SRC += $(TOPDIR)/src/$(MODULE)/share/native/common/awt/medialib
 endif
 
@@ -156,7 +156,7 @@
       #
 endif
 
-ifeq ($(call And, $(call isTargetOs, solaris) $(call isTargetCpu, sparcv9)), true)
+ifeq ($(call And, $(call isTargetOs, solaris) $(call isTargetCpu, not-really-sparcv9)), true)
   LIBAWT_EXFILES += java2d/loops/MapAccelFunc.c
 else
   LIBAWT_EXCLUDES += \
@@ -183,7 +183,7 @@
 
 LIBAWT_CFLAGS += -D__MEDIALIB_OLD_NAMES -D__USE_J2D_NAMES $(X_CFLAGS)
 
-ifeq ($(call And, $(call isTargetOs, solaris) $(call isTargetCpu, sparcv9)), true)
+ifeq ($(call And, $(call isTargetOs, solaris) $(call isTargetCpu, not-really-sparcv9)), true)
   LIBAWT_CFLAGS += -xarch=sparcvis -DMLIB_ADD_SUFF \
       $(TOPDIR)/src/$(MODULE)/unix/native/libmlib_image/vis_$(OPENJDK_TARGET_CPU_BITS).il
 
@@ -190,7 +190,7 @@
   LIBAWT_ASFLAGS = -P -xarch=v9a
 endif
 
-ifeq ($(call isTargetOs, solaris), false)
+ifeq ($(call isTargetOs, not-really-solaris), false)
   LIBAWT_CFLAGS += -DMLIB_NO_LIBSUNMATH
 endif
 
